include/RE/B/BSFaceGenManager.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "RE/B/BSFixedString.h"
+#include "RE/B/BSTSingleton.h"
+
+namespace RE
+{
+	class BGSHeadPart;
+	class BSFaceGenModel;
+	class BSFaceGenNiNode;
+	class TESNPC;
+
+	class BSFaceGenManager : public BSTSingletonSDM<BSFaceGenManager>
+	{
+	public:
+		struct Action
+		{
+			BSFixedString name;
+			std::uint32_t unk04;
+			float		  delta;
+		};
+
+		struct BSFaceGenModelMap
+		{
+		public:
+			struct Entry : public NiRefObject
+			{
+			public:
+				// members
+				NiPointer<BSFaceGenModel> model;  // 10
+				std::uint64_t			  time;	  // 18
+			};
+			static_assert(sizeof(Entry) == 0x20);
+
+			// members
+			BSTHashMap<BSFixedString, NiPointer<Entry>> map;			 // 00
+			BSReadWriteLock								lock;			 // 30
+			bool										hasFaceGenData;	 // 38
+		};
+		static_assert(sizeof(BSFaceGenModelMap) == 0x40);
```

Comment:
Can you place this in the `RE` namespace just before `BSFaceGenManager`?

Reply:
Sure